### ISSUE TITLE
simclr_loss test failing in GPU

### ISSUE THREAD

**Issue by tanzhenyu** [keras-team/keras-cv#793]
You: Running pytest keras_cv/losses/simclr_loss_test.py locally failed:
"def test_output_value(self):
AssertionError:
  Max relative difference:
    x: array(3.566533)
    y: array(3.566689)
"

This test passes in CPU.

**Comment by ianstenbit** [comment]
@tanzhenyu thanks for filing this issue.

Can you please add more specifics about the GPU platform / environment you're using?
These tests are passing on our GCB GPU tests, so I'll need to know more specifics to debug this.

Thanks!

**Comment by ianstenbit** [comment]
I've now run this test manually on the following devices:
- Apple M1 chip
- NVIDIA Tesla T4
- NVIDIA Tesla V100

and in each case I am getting a consistent loss of `3.566689`

I don't think I'll be able to debug further without access to your machine.

I do see that in the [Keras core losses tests](https://github.com/keras-team/keras/blob/master/keras/losses_test.py), many loss assertions have a tolerance of 1e-3, which is a wide enough tolerance that would account for this discrepancy.

I'm not sure if that's the right approach, but I don't see any clear issues in the code which might cause this.

**Comment by tanzhenyu** [comment]
> I've now run this test manually on the following devices:
> 
>     * Apple M1 chip
> 
>     * NVIDIA Tesla T4
> 
>     * NVIDIA Tesla V100
> 
> 
> and in each case I am getting a consistent loss of `3.566689`
> 
> I don't think I'll be able to debug further without access to your machine.
> 
> I do see that in the [Keras core losses tests](https://github.com/keras-team/keras/blob/master/keras/losses_test.py), many loss assertions have a tolerance of `1e-3`, which is a wide enough tolerance that would account for this discrepancy.
> 
> I'm not sure if that's the right approach, but I don't see any clear issues in the code which might cause this.

Yeah I realized that as well. It seems there is some numerical issues across different platforms. For now, I'm not sure whether this is due to the device type, or the driver version. Can you try the installing the same version for V100 and see if that makes a difference?

**Comment by tanzhenyu** [comment]
This looks like it's only failing on consumer hardware
